### PR TITLE
Submission 3

### PR DESCRIPTION
**Pull request by dimascatur**
this is my project for completing the task of submission, beg for the review.

### CODE REVIEW

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt [hunk 1]
```diff
@@ -59,8 +73,17 @@ class DetailMatchActivity : AppCompatActivity(),
             away_red.invisible()
             home_yellow.invisible()
             away_yellow.invisible()
+            date_match.invisible()
```
Comment (alfianyusufabdullah) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: Should be write like last our discussion.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt [hunk 2]
```diff
@@ -81,16 +105,95 @@ class DetailMatchActivity : AppCompatActivity(),
...
+        database.use {
+            val result = select(Favorite.TABLE_FAVORITE)
+                .whereArgs(
+                    "(MATCH_ID = {id})",
+                    "id" to matchDetail
+                )
+            val favorite = result.parseList(classParser<Favorite>())
+            if (favorite.isNotEmpty()) isFavorite = true
+        }
+    }
```
Comment (alfianyusufabdullah) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: Move to presenter.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt [hunk 3]
```diff
@@ -81,16 +105,95 @@ class DetailMatchActivity : AppCompatActivity(),
...
+        try {
+            database.use {
+                delete(
+                    Favorite.TABLE_FAVORITE, "(MATCH_ID = {id})",
+                    "id" to matchDetail
+                )
+            }
+        } catch (e: SQLiteConstraintException) {
+        }
+    }
```
Comment (alfianyusufabdullah) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: Move to presenter.

File: app/src/main/res/layout/activity_detail_match.xml [hunk 4]
```diff
@@ -234,6 +234,19 @@
         app:layout_constraintTop_toBottomOf="@+id/text_red"
         tools:text="69':Own  Jonjo Shelvey;61': Adam Lallana;51': Adam Lallana;33': Alberto Moreno;" />
```
Comment (alfianyusufabdullah) [inline app/src/main/res/layout/activity_detail_match.xml]: Replace hardcode value in **layout_height** and **layout_width** attirbute of `TextView` with `wrap_content`.

File: app/src/main/res/layout/item_player.xml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        android:contentDescription="@string/dummy"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:src="@drawable/player" />
+
+    <TextView
+        android:id="@+id/player_name"
+        android:layout_width="wrap_content"
+        android:layout_height="20dp"
+        android:textColor="@android:color/black"
```
Comment (alfianyusufabdullah) [inline app/src/main/res/layout/item_player.xml]: Replace hardcode value in **layout_height** and **layout_width** attirbute of `TextView` with `wrap_content`.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt (hunk 1 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: Resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt (hunk 2 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt (hunk 3 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: resolved, thanks

File: app/src/main/res/layout/item_search_match.xml [hunk 6]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (nrohmen) [inline app/src/main/res/layout/item_search_match.xml]: No need

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/favorite/MatchActivity.kt [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+class MatchActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_favorite)
+
+        val teamMatch = intent.getStringExtra(EXTRA_ID)
+
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/favorite/MatchActivity.kt]: Over spacing

**Pull request merged**